### PR TITLE
feat: create a payment

### PR DESCRIPTION
**Pull request by JoshuaOloton**
​

## Description

Implemented endpoint to post(create) a single payment

## Related Issue (Link to issue ticket)

Related to #446 

## Motivation and Context

This adds a new endpoint to the project to enable the creation of a new payment.

​

## How Has This Been Tested?

Unit tests were written and carried out for this endpoint using the Pytest library.

​
## Screenshots (if appropriate - Postman, etc):
image
image
image

​

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/payment.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from api.v1.schemas.token import TokenData
+from typing import List
+
+payment = APIRouter(prefix='/payments', tags=['Payment'])
+
+
+# @payment.get('/{payment_id}', status_code=status.HTTP_200_OK)
+# async def get_single_payment(payment_id: str, db: Session = Depends(get_db), current_user: User = Depends(user_service.get_current_user)):
+#     payment = PaymentService.fetch(db, payment_id)
+#     return payment
```
Comment (Goketech) [inline api/v1/routes/payment.py:19]: you commented this out

File: requirements.txt [hunk 2]
```diff
@@ -52,6 +52,6 @@ typer==0.12.3
 typing_extensions==4.12.2
 uuid7==0.1.0
 uvicorn==0.30.3
-uvloop==0.19.0
+# uvloop==0.19.0
```
Comment (Goketech) [inline requirements.txt:55]: revert changes made here

File: tests/v1/test_post_payment.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+#     mock_user = create_mock_user(db_session_mock)
+#     mock_payment = create_mock_payment(db_session_mock, mock_user.id)
+
+#     token = user_service.create_access_token(user_id=str(uuid7()))
+
+#     response = client.get(f"/api/v1/payments/{mock_payment.id}", headers={
+#         "Authorization": f"Bearer {token}"
+#     })
+
+#     assert response.status_code == 200
```
Comment (Goketech) [inline tests/v1/test_post_payment.py:93]: you commented this out too

File: api/v1/schemas/token.py [hunk 4]
```diff
@@ -9,7 +9,7 @@ class Token(BaseModel):
     token_type: str
 
 class TokenData(BaseModel):
-    username: Optional[str] = None
+    user_id: Optional[str] = None
```
Comment (Goketech) [inline api/v1/schemas/token.py:12]: you edited this schema and I don't see you making use of it

**Review by Goketech — CHANGES_REQUESTED** [review]

File: api/v1/routes/payment.py (hunk 1 above)
Comment (JoshuaOloton) [reply api/v1/routes/payment.py:19]: Yes I'm creating two endpoints for separate tickets, I'm pushing the commented sections in a separate PR

File: requirements.txt (hunk 2 above)
Comment (JoshuaOloton) [reply requirements.txt:55]: Alright, will do

File: api/v1/schemas/token.py (hunk 4 above)
Comment (JoshuaOloton) [reply api/v1/schemas/token.py:12]: Yes the schema had already been implemented in the user service, I can remove the schema in another commit

File: tests/v1/test_post_payment.py (hunk 3 above)
Comment (JoshuaOloton) [reply tests/v1/test_post_payment.py:93]: Yes thank you, its the same issue
I was assigned to get Payment by ID ticket and I had to pick up the Post payments as well, the person assigned to it was unable to deliver. I'll make the PR separately now

**Comment by JoshuaOloton** [comment]
Corrections have been made

File: api/v1/routes/payment.py (hunk 1 above)
Comment (bolexs) [reply api/v1/routes/payment.py:19]: @Goketech has this been resolved, please don't comment out code. Anything that is not been used should be removed

File: tests/v1/test_post_payment.py (hunk 3 above)
Comment (bolexs) [reply tests/v1/test_post_payment.py:93]: same here

**Review by Goketech — DISMISSED** [review]

**Comment by Goketech** [comment]
@joboy-dev

File: api/v1/routes/payment.py (hunk 1 above)
Comment (JoshuaOloton) [reply api/v1/routes/payment.py]: Yes it has been resolved already @bolexs

**Review by joboy-dev — APPROVED** [review]

**Review by Goketech — APPROVED** [review]

File: api/v1/routes/payment.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+from api.v1.models.user import User
+from api.v1.services.user import user_service
+from api.v1.schemas.payment import CreatePaymentSchema
+
+payment = APIRouter(prefix='/payments', tags=['Payment'])
+
+
+
+@payment.post('/create', status_code=status.HTTP_201_CREATED)
+async def post_single_payments(request: CreatePaymentSchema, db: Session = Depends(get_db), current_user: User = Depends(user_service.get_current_user)):
```
Comment (joboy-dev) [inline api/v1/routes/payment.py:15]: Change request to `payment_schema`

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Resolve your merge conflict and make changes

File: api/v1/routes/payment.py (hunk 5 above)
Comment (JoshuaOloton) [reply api/v1/routes/payment.py:15]: Pls clarify the exact change needed for `payment-schema`

Comment (JoshuaOloton) [reply api/v1/routes/payment.py]: this has been resolved @joboy-dev

**Pull request closed without merging**